src/main/java/qna/domain/Answer.java

```diff
@@ -29,45 +34,39 @@ public Answer(User writer, Question question, String contents) {
 
     public Answer(Long id, User writer, Question question, String contents) {
         super(id);
-
-        if(writer == null) {
+        if (writer == null) {
             throw new UnAuthorizedException();
         }
-
-        if(question == null) {
+        if (question == null) {
             throw new NotFoundException();
         }
-
         this.writer = writer;
         this.question = question;
         this.contents = contents;
     }
 
-    public Answer setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
-
     public boolean isDeleted() {
         return deleted;
     }
 
-    public boolean isOwner(User writer) {
-        return this.writer.equals(writer);
-    }
-
     public User getWriter() {
         return writer;
     }
 
-    public String getContents() {
-        return contents;
-    }
-
     public void toQuestion(Question question) {
         this.question = question;
     }
 
+    DeleteHistory delete() {
+        DeleteHistory history = DeleteHistory.answerOf(getId(), writer);
+        deleted = true;
+        return history;
```

Comment:
```suggestion
        return DeleteHistory.answerOf(getId(), writer);
```

여기서도 로직이 간단해서 지역 변수없이 바로 리턴 시켜볼 수 있겠네요.

Reply:
만약 DeliveryHistory 를 생성하다가 예외가 발생된다면 
상태가 바뀐 상태로 남아있고 예외가 던져지기 때문에 일부러 상태 변경을 뒤로 미루도록 설정했습니다.

Reply:
@devyonghee 
제가 답변 주신 부분을 늦게 보았네요 ㅠㅠ
저도 이부분에 대해서 리뷰하면서 고민해 보았는데
Spring을 사용하면 Transactional 애노테이션을 통해 deleted 상태도 롤백 되기 때문에
크게 영향이 없을거라 생각했습니다. 😄